src/actions/settingsActions.js

```diff
@@ -0,0 +1,14 @@
+import { SETTINGS_CHANGE } from '../constants/actionTypes'
+
+export function changeSettings (payload) {
+  return {
+    type: SETTINGS_CHANGE,
+    settings: payload,
+  }
+}
+
+export const setLocale = (newLocale) => changeSettings({locale: newLocale})
+export const setLayout = (layout) => changeSettings({layout: layout})
+export const setEventsColors = (enabled) => changeSettings({eventsColors: enabled})
+export const setFullWeek = (enabled) => changeSettings({fullWeek: enabled})
+export const setFacultyGrid = (enabled) => changeSettings({facultyGrid: enabled})
```

Comment:
Yeah, but I was thinking about exposing them all as actions for the relevant component to reduce coupling a bit, we can drop them later.